soda/scientific/soda/scientific/distribution/comparison.py

```diff
@@ -165,28 +171,45 @@ def evaluate(self) -> Tuple[float, float]:
             test_data_frequencies,
             ref_data_frequencies * np.mean(test_data_frequencies) / np.mean(ref_data_frequencies),
         )
-        return stat_value, p_value
-
-    def generate_ref_data(self) -> pd.Series:
-        sample_size = len(self.test_data)
-        return pd.Series(self.rng.choice(self.cfg.bins, p=self.cfg.weights, size=sample_size))
+        return dict(stat_value=stat_value, check_value=p_value)
 
 
-class KSAlgorithm(DistributionAlgorihm):
+class KSAlgorithm(DistributionAlgorithm):
     def __init__(self, cfg: RefDataCfg, test_data: pd.Series, seed: int = 61) -> None:
+        cfg.dtype = "continuous"
         super().__init__(cfg, test_data, seed)
 
-    def evaluate(self) -> Tuple[float, float]:
+    def evaluate(self) -> dict:
```

Comment:
better to have fully specified types. There are a few similar issues in the rest of the code. I won't point repeat them to not make the review too overwhelming but please go and change them so our types are a bit more useful